File: articles/can-grit-be-taught/c/can_grit_be_taught_effect_size.c
#include <stdio.h>
#include <math.h>

double cohens_d(double mean_treatment, double mean_control, double pooled_sd) {
    if (pooled_sd == 0.0) return 0.0;
    return (mean_treatment - mean_control) / pooled_sd;
}

int main(void) {
    double d = cohens_d(4.2, 3.6, 0.8);
    printf("Synthetic intervention effect size estimate: %.3f\n", d);
    printf("Professional caution: synthetic demonstration only.\n");
    return 0;
}

File: articles/can-grit-be-taught/rust/can_grit_be_taught_effect_size.rs
fn cohens_d(mean_treatment: f64, mean_control: f64, pooled_sd: f64) -> f64 {
    if pooled_sd == 0.0 {
        0.0
    } else {
        (mean_treatment - mean_control) / pooled_sd
    }
}

fn main() {
    let d = cohens_d(4.2, 3.6, 0.8);
    println!("Synthetic intervention effect size estimate: {:.3}", d);
    println!("Professional caution: synthetic demonstration only.");
}
